docs/conf.py

```diff
@@ -99,3 +105,27 @@
 
 # Add support for inline tabs
 extensions += ['sphinx_inline_tabs']
+
+# Add support for the unreleased "next-version" change notes
+extensions += ['sphinxcontrib.towncrier']
+
+
+# -- Options for towncrier_draft extension --------------------------------------------
+
+PROJECT_ROOT_DIR = Path(__file__).parents[1].resolve()
+get_scm_version = partial(get_version, root=PROJECT_ROOT_DIR)
+
+# The short X.Y version
+version = '.'.join(
+    get_scm_version(
+        local_scheme='no-local-version',
+    ).split('.')[:3],
+)
+
+# The full version, including alpha/beta/rc tags
+release = get_scm_version()
```

Comment:
This logic is redundant and conflicts with the work done by `jaraco.packaging.sphinx`. Is it needed?

Reply:
It's needed because otherwise `jaraco.packaging.sphinx` selects the hardcoded version of setuptools which is something that's already been released while this match makes it the next guess provided by setuptools-scm.

If I remove this, the changelog would look like:
```
# v56
- entries that are still in the form of fragments
# v56
- entries you've generated on release
```
but the goal of this change is to have something like
```
# v57.dev0
- entries that are still in the form of fragments
# v56
- entries you've generated on release
```

So that next version preview should either be generated here in sphinx or you should modify your release flow to change the hardcoded version to `v57.dev0` right after the release.

Here's what it looks like:
![setuptools-towncrier-draft](https://user-images.githubusercontent.com/578543/115994287-a2eda600-a5d6-11eb-83c7-a7f2628e5044.png)

Reply:
Here's the PR build for preview: https://setuptools--2650.org.readthedocs.build/en/2650/history.html

Reply:
I would actually like to get this project enabled with setuptools_scm, but until the project itself can adopt setuptools_scm, it seems a bad idea to paper over the issue that `setup.py --version` is reporting an incorrect version by partially adopting setuptools_scm but only in docs.

Currently, setuptools defines `egg_info.tag_build` and `egg_info.tag_date` to provide a post-release tag. I'm a little surprised that `setup.py --version` (what jaraco.packaging currently uses to elicit the version) doesn't include that tag. That's probably another bug.

Regardless, `jaraco.packaging` should probably invoke PEP 517 to get the package metadata and the version from that. Doing so would have the unfortunate side-effect of rendering the `.post` and date even for tagged commits (which are [only disabled when cutting a release](https://github.com/pypa/setuptools/blob/b5fa6ad11e1344648b470ff0d847a7d940b4c99d/tox.ini#L67-L68)).

Since adopting setuptools_scm for this project would solve all of these problems, I'd like to first tackle that issue (#2656).

Reply:
Alright, I'll change the approach a bit and will just remove the version from this title, for the time being, it's not that important. We can read it later, once #2656 is in.